waku/v2/protocol/waku_filter_v2/protocol.nim

```diff
@@ -157,3 +204,24 @@ proc new*(T: type WakuFilter,
   )
   wf.initProtocolHandler()
   wf
+
+const MaintainSubscriptionsInterval* = 1.minutes
+
+proc startMaintainingSubscriptions*(wf: WakuFilter, interval: Duration) =
```

Comment:
beyond this pr, not blocking.

since we have a few of these timers, I would suggest creating a [heartbeat](https://github.com/status-im/nim-libp2p/blob/c11772c94e44f0987b79d6593f34d6a4f92f07c5/libp2p/utils/heartbeat.nim#L20) so that we can reuse code.

and then just.

```
heartbeat "sometext", interval:
  await functionToRun.
```

ofc it depends if:
* we want to run it exactly every x time.
* or wait x time between the last completed run.